### PR TITLE
Menu

### PR DESCRIPTION
**Pull request by MortenSOlesen**
### Added the following files
- menus.c
- TUI_Examples
  - TUI.txt
  - TUI_TIme.txt
  - TUI_Appliances.txt
- appliance.h
- appliances.h

### To add a menu
add a call name under: `enum MENU`
add menu information to: `menu menus array`
add a case to: `exec_menu`
add a function containing your cases similar to: `start_function`
*a case might run a piece of code, go to another menu etc.*

### CODE REVIEW

File: PowerHouse/appliance.h [hunk 1]
```diff
@@ -0,0 +1,7 @@
+typedef struct Appliance
+{
+    char name[50]; //Appliance name 
+    double wh; // watt hours used
```
Comment (theis999) [inline PowerHouse/appliance.h]: Watt hours already used or is it the cost of running a cycle?

File: PowerHouse/appliance.h [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+    double runTime; // time to run appliance rounded to whole hour
+
+} Appliance;
```
Comment (theis999) [inline PowerHouse/appliance.h]: Please add tracing linebreak so github doesn't complain
image

File: PowerHouse/appliances.h [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#include "appliance.h"
+
+int ApplianceCount = 3;
+
+Appliance Appliances[50] = {
```
Comment (theis999) [inline PowerHouse/appliances.h:5]: Is it possible to reuse the ApplianceCount as the array init size?

File: PowerHouse/menus.c [hunk 4]
```diff
@@ -0,0 +1,174 @@
+#include <stdio.h>
+#include <stdlib.h>
+
```
Comment (theis999) [inline PowerHouse/menus.c:3]: no include of appliances.h? is that intentional

File: PowerHouse/menus.c [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+typedef enum MENU
+{
+    start,
+    time,
+    carbon,
+    get24hStatus
+} MENU;
+
+int run = 1;
```
Comment (theis999) [inline PowerHouse/menus.c]: Perhaps rename to running

File: PowerHouse/menus.c [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+int MenuID = start;
+
+typedef struct menu
+{
+    enum MENU menuID;                              // Unique ID
```
Comment (theis999) [inline PowerHouse/menus.c]: Id or type should be sufficient as a variable name

**Review by theis999 — APPROVED** [review]
There is nothing that looks wrong. However the inclusion of appliance.h & appliances.h in the commit is confusing as they do not appear to be used.

File: PowerHouse/appliance.h (hunk 1 above)
Comment (MortenSOlesen) [reply PowerHouse/appliance.h]: cost of running cycle, comment edited

File: PowerHouse/menus.c (hunk 4 above)
Comment (MortenSOlesen) [reply PowerHouse/menus.c:3]: appliances.h is not part of menus (altho its he same commit x) ), so yes

File: PowerHouse/appliances.h (hunk 3 above)
Comment (MortenSOlesen) [reply PowerHouse/appliances.h:5]: The user can add extra `Appliance`(s), hence the appliance count, `Appliances[i]` is the max possible allowed.
`ApplianceCount` is used to keep track of how many are currently added so we do not run the entire array through.

**Pull request merged**